cpp/include/cudf/functions.h

```diff
@@ -1111,6 +1101,60 @@ gdf_error gdf_quantile_approx(gdf_column* col_in,
                               gdf_scalar*  result,
                               gdf_context* ctxt);
 
+/**
+ * @brief Sorts a set of columns based on specified "key" columns. Returns a column containing
+ * the offset to the start of each group.
+ *
+ * @param[in]  num_data_cols            The number of columns in the dataset.
```

Comment:
@jrhemstad should the input set of columns be `cudf::table`s?

Reply:
Ideally, yes. Otherwise we'll just need to come back and fix this later. Would prefer that it just be done the right way up front.

@jeanp413 see https://gist.github.com/jrhemstad/717054db12c3755a8c4a7f40647ddfb6

Reply:
done